test/e2e-common.sh

```diff
@@ -43,6 +43,10 @@ INSTALL_MONITORING=0
 INSTALL_CUSTOM_YAMLS=""
 
 UNINSTALL_LIST=()
+readonly TMP_DIR=$(mktemp -d -t ci-$(date +%Y-%m-%d-%H-%M-%S)-XXXXXXXXXX)
```

Comment:
If the function call on the right side failed, readonly swallows the exit code and still return 0
```suggestion
TMP_DIR=$(mktemp -d -t ci-$(date +%Y-%m-%d-%H-%M-%S)-XXXXXXXXXX)
readonly TMP_DIR
```

Reply:
```
    --system-namespace)
      [[ -z "$2" ]] || [[ $2 = --* ]] && fail_test "Missing argument to --system-namespace"
      readonly E2E_SYSTEM_NAMESPACE=$2
      return 2
```
This param is set to readonly in later code.